pkg/objstore/s3/s3.go

```diff
@@ -307,6 +309,53 @@ func (b *Bucket) Name() string {
 	return b.name
 }
 
+// Reads the config file and changes the credentials of the instance if yaml is changed.
+func (b *Bucket) ReloadCredentials() error {
+	if len(b.confFilepath) > 0 {
+		c, err := ioutil.ReadFile(b.confFilepath)
+		if err != nil {
+			return errors.Errorf("Error reading config file.")
+		}
+		config, err := parseConfig(c)
+		if err != nil {
+			return err
+		}
+		if err := validate(config); err != nil {
+			return err
+		}
+		var chain []credentials.Provider
+		wrapCredentialsProvider := func(p credentials.Provider) credentials.Provider { return p }
+		if config.AccessKey != "" {
+			chain = []credentials.Provider{wrapCredentialsProvider(&credentials.Static{
+				Value: credentials.Value{
+					AccessKeyID:     config.AccessKey,
+					SecretAccessKey: config.SecretKey,
+					SignerType:      credentials.SignatureV4,
+				},
+			})}
+			var rt http.RoundTripper
+			if config.HTTPConfig.Transport != nil {
+				rt = config.HTTPConfig.Transport
+			} else {
+				rt = DefaultTransport(config)
+			}
+			client, err := minio.New(config.Endpoint, &minio.Options{
+				Creds:     credentials.NewChainCredentials(chain),
+				Secure:    !config.Insecure,
+				Region:    config.Region,
+				Transport: rt,
+			})
+			if err != nil {
+				return errors.Wrap(err, "initialize s3 client.")
+			}
+			(*b).client = client
```

Comment:
```suggestion
			(b).client = client
```
More Info - https://tour.golang.org/moretypes/4